hkl/configuration.py

```diff
@@ -22,51 +31,274 @@
+    """(internal) Raise ValueError if value is not between low & high."""
+    if low > high:
+        raise ValueError(f"{intro}:  {low} should not be greater than {high}")
+    if not (low <= value <= high):
+        raise ValueError(f"{intro}:  {value} is not between {low} & {high}")
+
+
+def _check_type(actual, expected, intro):
+    """(internal) Raise TypeError if actual is not an instance of expected."""
     if not isinstance(actual, expected):
-        raise TypeError(f"{words}:  received: {actual}  expected: {expected}")
+        raise TypeError(f"{intro}:  received: {actual}  expected: {expected}")
 
 
-def _check_value(actual, expected, words):
-    """Raise ValueError if actual is not equal to expected."""
+def _check_value(actual, expected, intro):
+    """(internal) Raise ValueError if actual is not equal to expected."""
     if actual != expected:
-        raise ValueError(f"{words}:  received: {actual}  expected: {expected}")
+        raise ValueError(f"{intro}:  received: {actual}  expected: {expected}")
 
 
 @dataclass
+class DCConstraint:
+    """(internal) Configuration of one diffractometer axis constraint."""
+
+    low_limit: float
+    high_limit: float
+    value: float
+    fit: bool
+
+    @property
+    def values(self):
+        """Return the list of values in order."""
+        return list(self.__dict__.values())
+
+    def validate(self, cname):
+        """
+        Check this constraint has values the diffractometer can accept.
+
+        Assumes diffractometer real axis limits of AX_MIN to AX_MAX degrees.
+        """
+        _check_range(self.low_limit, AX_MIN, AX_MAX, f"{cname} low_limit")
+        _check_range(self.high_limit, AX_MIN, AX_MAX, f"{cname} high_limit")
+        _check_range(self.value, AX_MIN, AX_MAX, f"{cname} value")
+        # no additional validation needed for 'fit'
+
+
+@dataclass
+class DCLattice:
+    """(internal) Configuration of one crystal lattice."""
+
+    a: float
+    b: float
+    c: float
+    alpha: float
+    beta: float
+    gamma: float
+
+    def validate(self, dc_obj):
+        """Check this lattice has values the diffractometer can accept."""
+        _check_range(self.a, 1e-6, 1e6, "a")
+        _check_range(self.b, 1e-6, 1e6, "b")
+        _check_range(self.c, 1e-6, 1e6, "c")
+        _check_range(self.alpha, 1e-6, 180.0 - 1e-6, "alpha")
+        _check_range(self.beta, 1e-6, 180.0 - 1e-6, "beta")
+        _check_range(self.gamma, 1e-6, 180.0 - 1e-6, "gamma")
+        # exclude zero
+        _check_not_value(self.alpha, 0.0, f"alpha")
+        _check_not_value(self.beta, 0.0, f"beta")
+        _check_not_value(self.gamma, 0.0, f"gamma")
+
+    @property
+    def values(self):
+        """Return the list of values in order."""
+        return list(self.__dict__.values())
+
+
+@dataclass
+class DCReflection:
+    """(internal) Configuration of one orientation reflection."""
+
+    reflection: dict[str, float]  # reciprocal-space axis positions
```

Comment:
FYI, there's a special syntax sphinx will pick up for attribute-level comments. It looks like this:
```suggestion
    #: Reciprocal-space axis positions
    reflection: dict[str, float]
```